### PR TITLE
Fix the tear down function in integration.modules.test_grains

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
`integration.modules.test_grains.GrainsAppendTestCase`
Fix the `tearDown` function to just set the grain to an empty list instead of removing each individual item.

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/46351

### Tests written?
Yes

### Commits signed with GPG?
Yes

### CODE REVIEW

File: tests/integration/modules/test_grains.py [hunk 1]
```diff
@@ -152,8 +152,7 @@ class GrainsAppendTestCase(ModuleCase):
     GRAIN_VAL = 'my-grain-val'
 
     def tearDown(self):
-        for item in self.run_function('grains.get', [self.GRAIN_KEY]):
-            self.run_function('grains.remove', [self.GRAIN_KEY, item])
```
Comment (cachedout) [inline tests/integration/modules/test_grains.py]: OK, but why is this necessary? I don't understand what's being fixed here. Is `grains.remove` not doing the right thing?

Comment (twangboy) [reply tests/integration/modules/test_grains.py]: In rare cases the `grains.get` function was returning a boolean value and the following stack trace was thrown:
```
Traceback (most recent call last):
  File "c:\testing\tests\integration\modules\test_grains.py", line 155, in tearDown
    for item in self.run_function('grains.get', [self.GRAIN_KEY]):
TypeError: 'bool' object is not iterable
```
So rather than loop over the grains just to remove each item, we decided to just set it to an empty list.

Comment (rallytime) [reply tests/integration/modules/test_grains.py]: @cachedout Can you swing back around here when you get a moment?

**Review by cachedout — APPROVED** [review]
